extruct/xmldom.py

```diff
@@ -53,7 +56,7 @@ def lxmlDomNodeType(node):
         return Node.NOTATION_NODE
 
 
-class DomHtmlMixin(object):
+class DomHtmlMixin:
```

Comment:
mixins can be typed in python but it's a little verbose and doesn't seem key for this first pass

I think getting the top level user facing API properly typed is the primary goal of this PR